From 9eac82b93debf84c57e010fa151623976895a42c Mon Sep 17 00:00:00 2001
From: Ronnie Olsson <ronnie.olsson@inter.ikea.com>
Date: Tue, 12 Oct 2021 11:19:27 +0200
Subject: [PATCH] [HSPP-7978] Read out hw-conf in fastboot

Adds getvar hw-conf to get hardware configuration.
1 is old variant with ST-Safe on i2c4
2 is new variant with ST-Safe on i2c6
---
 drivers/fastboot/fb_getvar.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/drivers/fastboot/fb_getvar.c b/drivers/fastboot/fb_getvar.c
index c849858d5c..af00d5b055 100644
--- a/drivers/fastboot/fb_getvar.c
+++ b/drivers/fastboot/fb_getvar.c
@@ -32,6 +32,7 @@ static void getvar_partition_size(char *part_name, char *response);
 static void getvar_is_userspace(char *var_parameter, char *response);
 static void getvar_gateway_id(char *gateway_id, char *response);
 static void getvar_rtc(char *gateway_id, char *response);
+static void getvar_hwconf(char *gateway_id, char *response);
 
 static const struct {
 	const char *variable;
@@ -88,6 +89,9 @@ static const struct {
 	}, {
 		.variable = "rtc",
 		.dispatch = getvar_rtc
+	}, {
+		.variable = "hw-conf",
+		.dispatch = getvar_hwconf
 	}
 };
 
@@ -302,6 +306,21 @@ static void getvar_rtc(char *rtc, char *response)
 		fastboot_response("OKAY", response, "%d", 2);
 }
 
+static void getvar_hwconf(char *gateway_id, char *response)
+{
+	char cmdbuf[32];
+
+	snprintf(cmdbuf, sizeof(cmdbuf), "i2c dev 2");
+	if (run_command(cmdbuf, 0))
+		fastboot_fail("IO", response);
+
+	snprintf(cmdbuf, sizeof(cmdbuf), "i2c probe 0x10");
+	if (run_command(cmdbuf, 0))
+		fastboot_response("OKAY", response, "%d", 1);
+	else
+		fastboot_response("OKAY", response, "%d", 2);
+}
+
 /**
  * fastboot_getvar() - Writes variable indicated by cmd_parameter to response.
  *
-- 
2.17.1

